### PR TITLE
[MIG][12.0] pos_order_to_sale_order (from 8.0)

### PR DESCRIPTION
**Pull request by legalsylvain**
- migrate ``pos_order_to_sale_order`` module from 8.0 to 12.0
- allow user to create sale orders from PoS with a new button

pos_frontend_order

Several options are so available : 

pos_frontend_buttons

CC : @quentinDupont 

#GRAPOCA

### CODE REVIEW

**Review by rousseldenis — APPROVED** [review]
Functional review. Works as expected

**Review by hparfr — CHANGES_REQUESTED** [review]
Please take in account prior work on this field from v10.
https://github.com/OCA/pos/pull/264

**Comment by legalsylvain** [comment]
Hi @hparfr. I saw the pending PR #264 when I began to work on this PR. Well #264 :
- is opened since 2 years and has no activity since almost 1 year
- has 2 "need fixing" review and 0 positive one
- Travis is red, while 10.0 branch is green (https://travis-ci.org/OCA/pos/branches)
- Runbot is also red

so, it is still WIP and seems abandonned

Fixing this PR was more work than migrating a clean and reviewed v8.0 work.

kind regards.

File: pos_order_to_sale_order/views/view_pos_config.xml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+                </div>
+                <div class="row">
+                    <label string="Create Confirmed Sale Orders" for="iface_create_confirmed_sale_order" class="col-lg-3 o_light_label"/>
+                    <field name="iface_create_confirmed_sale_order"/>
+                </div>
+                <div class="row">
+                    <label string="Create Delivered Sale Orders" for="iface_create_delivered_sale_order" class="col-lg-3 o_light_label"/>
+                    <field name="iface_create_delivered_sale_order"/>
+                </div> -->
+
```
Comment (ivantodorovich) [inline pos_order_to_sale_order/views/view_pos_config.xml]:
```suggestion
```

File: pos_order_to_sale_order/static/src/xml/pos_order_to_sale_order.xml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+                                id="button-create-delivered-order" action="delivered">
+                            Create Delivered Sale Order
+                        </div>
+                    </div>
+                </div>
+            </div>
+        </div>
+     </t>
+<!-- t-att-data-action="'confirmed'" -->
+
```
Comment (ivantodorovich) [inline pos_order_to_sale_order/static/src/xml/pos_order_to_sale_order.xml]:
```suggestion
```

File: pos_order_to_sale_order/static/src/js/screens.js [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+
+           this.$('.paymentmethod').click(function(event){
+                self.click_sale_order_button(event.currentTarget.attributes.action.nodeValue);
+            });
+        },
+
+        click_sale_order_button: function(action) {
+            var self = this;
+            console.log("click_sale_order_button : " + action);
+
```
Comment (ivantodorovich) [inline pos_order_to_sale_order/static/src/js/screens.js]:
```suggestion
```

**Review by ivantodorovich — APPROVED** [review]
Minor comments, otherwise LGTM :+1:

**Pull request merged**